src/sentry/models/outbox.py

```diff
@@ -570,10 +570,12 @@ def process_shard(
             yield next_shard_row
 
     @contextlib.contextmanager
-    def process_coalesced(self) -> Generator[OutboxBase | None, None, None]:
+    def process_coalesced(
+        self, is_synchronous_flush: bool
+    ) -> Generator[OutboxBase | None, None, None]:
         coalesced: OutboxBase | None = self.select_coalesced_messages().last()
         first_coalesced: OutboxBase | None = self.select_coalesced_messages().first() or coalesced
-        tags = {"category": "None"}
+        tags: dict[str, int | str] = {"category": "None", "synchronous": int(is_synchronous_flush)}
```

Comment:
Not sure if I'm missing some nuance here, but mypy was complaining about this type without the type hint.

`src/sentry/models/outbox.py:585: error: Argument "tags" to "timing" has incompatible type "Dict[str, object]"; expected "Optional[Mapping[str, Union[str, int, float, None]]]"  [arg-type]`

Any ideas why this was being type inferred as `Dict[str, object]`?

Reply:
> Not sure if I'm missing some nuance here, but mypy was complaining about this type without the type hint.

I've run into this a few times with dictionaries that are incrementally built as it sees that the values aren't all one type and assumes the worst.